allennlp/commands/fine_tune.py

```diff
@@ -215,12 +215,20 @@ def fine_tune_model(model: Model,
     for name in tunable_parameter_names:
         logger.info(name)
 
-    trainer = Trainer.from_params(model,
-                                  serialization_dir,
-                                  iterator,
-                                  train_data,
-                                  validation_data,
-                                  trainer_params)
+    # If the config specifies a trainer subclass, we should use it.
+    # There must exist a trainer.
+    trainer_params = params.get("trainer")
+    trainer_choice = trainer_params.pop_choice("type",
+                                               Trainer.list_available(),
+                                               True)
+    trainer_cls = Trainer.by_name(trainer_choice)
+
+    trainer = trainer_cls.from_params(model,
```

Comment:
I'd be inclined to combine this line with the one above it, so you have `Trainer.by_name(trainer_choice).from_params(...)`

Reply:
I would call this function using keyword arguments

`from_params(model=model, ... )`

otherwise you're implicitly requiring any trainer subclass to have a from_params that takes the exact same arguments in the exact same order, which seems unnecessary

Reply:
Got it. Good suggestions. Thanks.